.github/workflows/invalidatePRs.sh

```diff
@@ -0,0 +1,137 @@
+#!/bin/bash
+
+# GitHub workflow handles invalidating Pull Requests after 3 hours.
+#
+# It will only invalidate open, non-draft pull requests and mark each status
+# as timed out.
+
+if [[ "" == $token ]]; then
+  echo Missing the environment variable "$token"
+  exit
+fi
+
+oauth=$token
+shouldDebug=false
+
+if [[ "" != $debug ]]; then
+  shouldDebug=true
+  echo Initializing debug log level
+fi
+
+# Get all active PRs
+# https://docs.github.com/en/rest/reference/pulls#list-pull-requests
+prs=$(curl -s \
+  -H "Accept: application/vnd.github.v3+json" \
+  -H "Authorization: token $oauth" \
+  https://api.github.com/repos/imodeljs/imodeljs/pulls)
+
+_jq() {
+  echo ${pr} | base64 --decode | jq -r ${1}
+}
+
+_jq2() {
+  echo ${status} | base64 --decode | jq -r ${1}
+}
+
+_jq3() {
+  echo ${1} | base64 --decode | jq -r ${2}
+}
+
+function log() {
+  if [[ $shouldDebug == true  ]]; then
+    echo "${1}"
+  fi
+}
+
+# Get head sha on master branch
+masterCommit=$(curl -s \
+  -H "Accept: application/vnd.github.v3+json" \
+  -H "Authorization: token $oauth" \
+  https://api.github.com/repos/imodeljs/imodeljs/commits/master)
+
+masterHeadCommit=$(echo ${masterCommit} | jq -r '. | @base64')
+masterSha=$(_jq3 ${masterHeadCommit} '.sha')
+echo ${masterSha}
+
+for pr in $(echo "${prs}" | jq -r '.[] | @base64'); do
+  #echo ${pr} | base64 --decode
+  log "$(_jq '.title') (#$(_jq '.number'))"
+
+  if [[ "open" != $(_jq '.state') ]] || [[ "true" == $(_jq '.draft') ]]; then
+    log "  Skipping due to state=$(_jq '.state') and draft=$(_jq '.draft')."
+    continue
+  fi
+
+  ref=$(_jq '.base.ref')
+  sha=$(_jq '.base.sha')
+
+  if [[ "master" != $ref ]]
```

Comment:
I don't think you need to special case `master` here, right? If you just treat `$ref` always as the base branch name to compare  `$sha`, which is the current position of the PR branch on ref, against the `$refSha` from below that should be enough.